### PR TITLE
Switch to CurseForge Core API

### PR DESCRIPTION
**Pull request by ryanccn**

### CODE REVIEW

**Comment by glowiak** [comment]
.vscode

**Review by timoreo22 — APPROVED** [review]
LGTM

**Review by d-513 — APPROVED** [review]

**Comment by d-513** [comment]
commit history will be pain btw. I suggest rebase your normal PR on stable one to have same commits.

**Comment by Scrumplex** [comment]
> commit history will be pain btw. I suggest rebase your normal PR on stable one to have same commits.

after merging this, we can just merge stable back into develop

**Comment by DioEgizio** [comment]
> commit history will be pain btw. I suggest rebase your normal PR on stable one to have same commits.

Wdym this is based on stable, we can just merge stable into develop

**Comment by d-513** [comment]
ok

**Comment by d-513** [comment]
no backport

**Comment by Scrumplex** [comment]
I would have preferred if formatting wasn't changed in this, or at least if it were in a seperate commit

File: launcher/modplatform/flame/FlameModIndex.cpp [hunk 1]
```diff
@@ -45,18 +34,22 @@ void FlameMod::loadIndexedPackVersions(ModPlatform::IndexedPack& pack,
...
+        if (versionArray.isEmpty()) {
+            continue;
+        }
 
         ModPlatform::IndexedVersion file;
         for (auto mcVer : versionArray) {
-            file.mcVersion.append(mcVer.toString());
+            auto str = mcVer.toString();
+
+            if (str.indexOf('.') > -1)
```
Comment (Scrumplex) [inline launcher/modplatform/flame/FlameModIndex.cpp]: nit
```suggestion
            if (str.contains('.'))
```

**Review by Scrumplex — COMMENTED** [review]
looks fine from my initial review.
No code changes are needed from my POV for 1.2.2

File: launcher/modplatform/flame/FlameModIndex.cpp [hunk 2]
```diff
@@ -45,18 +34,22 @@ void FlameMod::loadIndexedPackVersions(ModPlatform::IndexedPack& pack,
...
+                file.mcVersion.append(mcVer.toString());
```
Comment (flowln) [inline launcher/modplatform/flame/FlameModIndex.cpp]:
```suggestion
                file.mcVersion.append(str);
```

**Review by flowln — APPROVED** [review]
There's some other stuff that could be a little better (a lot of `QJsonArray.size() < 1` that could be `QJsonArray.isEmpty()`, for instance), but those were already there before your changes, so it's fine!

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**